tests/sentry/digests/backends/test_redis.py

```diff
@@ -0,0 +1,307 @@
+from __future__ import absolute_import
+
+import functools
+import itertools
+import mock
+import time
+
+from exam import fixture
+from redis.client import StrictRedis
+
+from sentry.digests import (
+    Record,
+)
+from sentry.digests.backends.redis import (
+    SCHEDULE_STATE_READY,
+    SCHEDULE_STATE_WAITING,
+    RedisBackend,
+    ensure_timeline_scheduled,
+    make_digest_key,
+    make_iteration_key,
+    make_record_key,
+    make_schedule_key,
+    make_timeline_key,
+    truncate_timeline,
+)
+from sentry.testutils import TestCase
+
+
+def get_set_size(cluster, key):
+    results = []
+    with cluster.all() as client:
+        results = client.zcard(key)
+    return sum(results.value.values())
+
+
+class BaseRedisBackendTestCase(TestCase):
+    DEFAULT_BACKEND_OPTIONS = {
+        'cluster': {
+            'hosts': {
+                0: {'db': 9},
+            },
+        },
+    }
+
+    def get_backend(self, options={}):
+        kwargs = self.DEFAULT_BACKEND_OPTIONS.copy()
```

Comment:
I don't think that's true (this only does a shallow merge, not a deep one, and doesn't mutate that later.)